### PR TITLE
Add JSON source

### PR DESCRIPTION
**Pull request by jhoekx**
The JSON source allows reading offline exports of sources using Kukur.

### CODE REVIEW

File: kukur/source/json/json.py [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+
+def _read_metadata(selector: SeriesSelector, path: Path) -> Metadata:
+    with path.open("rb") as f:
+        data = json.load(f)
+        return Metadata.from_data(data, selector)
+
+
+def from_config(config: Dict) -> JSONSource:
+    """Create a new JSON source."""
+    if "metadata_path" not in config:
```
Comment (WouterWouters) [inline kukur/source/json/json.py:61]: Should be `metadata` or adjusted in asciidoc and below.

File: tests/test_data/json/json-examples.toml [hunk 2]
```diff
@@ -0,0 +1,5 @@
+[source.json]
+type = "json"
+metadata_path = "tests/test_data/json"
```
Comment (WouterWouters) [inline tests/test_data/json/json-examples.toml:3]: Can be removed?

**Review by WouterWouters — CHANGES_REQUESTED** [review]

File: tests/test_data/json/json-examples.toml (hunk 2 above)
Comment (jhoekx) [reply tests/test_data/json/json-examples.toml]: Indeed. Done.

File: kukur/source/json/json.py (hunk 1 above)
Comment (jhoekx) [reply kukur/source/json/json.py]: Correct. Was a last-minute change to be consistent with the other sources.

Changed.

**Review by WouterWouters — APPROVED** [review]

**Comment by jhoekx** [comment]
@hammsvietro to validate

**Comment by hammsvietro** [comment]
Working correctly.
Adapted an existing source to use JSON metadata, produced the same results as csv.

**Pull request merged**